src/Analyser/NodeScopeResolver.php

```diff
@@ -951,7 +951,10 @@ private function processStmtNode(
 						]),
 					)));
 				} else {
```

Comment:
I'm not sure about the purpose of 
```
if ($this->polluteScopeWithAlwaysIterableForeach) {
     $finalScope = $finalScope->mergeWith($scope->filterByTruthyValue(new BooleanOr(
	     new BinaryOp\Identical(
		     $stmt->expr,
			 new Array_([]),
	     ),
		 new FuncCall(new Name\FullyQualified('is_object'), [
			 new Arg($stmt->expr),
		 ]),
	)));
} else {
	$finalScope = $finalScope->mergeWith($scope);
}
```

But I soon as I make this change, the issue is solved.

I dunno
- if it has a bad impact (I can add tests if you had some in mind)
- if I should just remove the else part and the if condition.